### PR TITLE
Updated the default scopes of the client credentials grant

### PR DESCRIPTION
**Pull request by clintonb**
Fixes issue introduced by #25.

ECOM-4196

### CODE REVIEW

**Comment by clintonb** [comment]
@nasthagiri please review. The client credentials grant type doesn't have a grant, like the other grant types, so there is nothing to check to ensure the client is not requesting greater access than is actually permitted. The client can request any scopes. Ultimately, the provider needs to enforce the limitation for either all, or specific, clients.

File: provider/oauth2/forms.py [hunk 1]
```diff
@@ -340,10 +340,15 @@ def clean(self):
...
+    def clean_scope(self):
+        # NOTE (CCB): This is a horrible hack, like much of our OAuth work. The scopes are declared in
+        # edx-oauth2-provider. (See edx_oauth2_provider/constants.py.) However, we need to provide a default scope
+        # that (a) gives the token basic read access and (b) allows access to the user info endpoint. This value
+        # represents the following scopes: openid (1), profile (2), email (4), permissions (32). At present, this is
+        # all scopes except course_staff and course_instructor. These scopes are normally associated with actual
+        # users, whereas the client credentials grant will primarily be used by service users.
+        #
+        # In the future, we should limit the allowable scopes either at a global or per-client level.
+        return 39
```
Comment (jcdyer) [inline provider/oauth2/forms.py:354]: Why not use the actual scopes instead of a magic number with a lengthy comment describing it?

```
return constants.OPENID | constants.PROFILE | constants.EMAIL | constants.PERMISSION
```

Are we trying to avoid making edx-django-oauth2-provider dependent on edx-oauth2-provider?

Comment (clintonb) [inline provider/oauth2/forms.py:354]: Yes, we end up with circular dependencies.

File: provider/oauth2/tests.py [hunk 2]
```diff
@@ -634,9 +639,10 @@ def assert_valid_access_token_response(self, access_token, response):
     def get_latest_access_token(self):
         return AccessToken.objects.filter(client=self.get_client()).order_by('-id')[0]
 
-    def test_authorize_success(self):
+    @ddt.data(None, 'read')
```
Comment (nasthagiri) [inline provider/oauth2/tests.py:642]: Can you use `constants.READ` here instead?
Also, didn't these 2 tests (`read` and `None`) work even before this PR?

File: provider/oauth2/forms.py [hunk 3]
```diff
@@ -340,10 +340,15 @@ def clean(self):
...
+    scope = forms.CharField(required=False)
 
-    def clean(self):
-        cleaned_data = super(ClientCredentialsGrantForm, self).clean()
-        # We do not fully support scopes for this grant type; however, a scope is required
-        # in order to create an access token. Default to read-only access.
-        cleaned_data['scope'] = constants.READ
-        return cleaned_data
+    def clean_scope(self):
+        # NOTE (CCB): This is a horrible hack, like much of our OAuth work. The scopes are declared in
```
Comment (nasthagiri) [inline provider/oauth2/forms.py:346]: Is this a temporary hack that will be cleaned up once we migrate completely to DOT?
